app/models/event.rb

```diff
@@ -49,14 +49,8 @@ def attendance_options
   def capacity(person_or_type = nil)
     overall_capacity = read_attribute(:capacity)
     return overall_capacity if person_or_type.nil?
-    klass = person_or_type.is_a?(Person) ? person_or_type.class.to_s : person_or_type.to_s
-    if (klass.eql? "Student") || (klass.eql? "Participant")
-      custom_capacity = student_capacity
-    elsif klass.eql? "Volunteer"
-      custom_capacity = volunteer_capacity
-    elsif klass.eql? "Mentor"
-      custom_capacity = mentor_capacity
-    end    
+    aud = Event.process_audience(person_or_type)
+    custom_capacity = attribute_for_audience(:capacity, aud)
     (custom_capacity.nil? || custom_capacity <= 0) ? overall_capacity : custom_capacity
```

Comment:
Why not build this logic and the "overall_capacity" stuff above into `attribute_for_audience`?